File: Java/DesignPatterns/src/main/java/designPatterns/behavioral/visitor/ConcretePart.java
package designPatterns.behavioral.visitor;

public class ConcretePart {
  static class A implements Part {
    @Override
    public String accept(Visitor visitor) {
      return visitor.visit(this);
    }
  }

  static class B implements Part {
    @Override
    public String accept(Visitor visitor) {
      return visitor.visit(this);
    }
  }

  static class C implements Part {
    @Override
    public String accept(Visitor visitor) {
      return visitor.visit(this);
    }
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/behavioral/visitor/ConcretePart.kt
package behavioral.visitor

sealed class ConcretePart {
  class A : Part {
    override fun accept(visitor: Visitor): String = visitor.visit(this)
  }

  class B : Part {
    override fun accept(visitor: Visitor): String = visitor.visit(this)
  }

  class C : Part {
    override fun accept(visitor: Visitor): String = visitor.visit(this)
  }
}
